client/src/featureform/format.py

```diff
@@ -20,6 +31,98 @@ def format_rows(format_obj, format_obj_2=None, format_obj_3=None, format_obj_4=N
     else:
         print(five_row_spacing.format(format_obj, format_obj_2, format_obj_3, format_obj_4, format_obj_5))
 
+
 def format_pg(s=""):
     print(divider)
     print(s)
+
+
+def display_statuses(stub, resources):
+    @dataclass
+    class Status:
+        resource_type: Type
+        name: str
+        variant: str
+        status: str
+        error: str
+
+    def get_statuses() -> List[Status]:
+        statuses = []
+        resources_to_check = {Feature, TrainingSet, Label, Source, Provider}
+        filtered_resources = filter(lambda r: type(r) in resources_to_check, resources)
+        for r in filtered_resources:
+            if r.name == "local-model":
+                continue
+            resource = r.get(stub)
+            variant = getattr(resource, "variant", "")
+            status = Status(
+                resource_type=type(r),
+                name=resource.name,
+                variant=variant,
+                status=resource.status,
+                error=resource.error,
+            )
+            statuses.append(status)
+        return statuses
+
+    def is_finished(statuses):
+        return all(
+            s.status == "READY"
+            or s.status == "FAILED"
+            or (s.resource_type is Provider and s.status == "NO_STATUS")
+            for s in statuses
+        )
+
+    print()
+    console = Console()
+    with Live(console=console, auto_refresh=True, screen=False) as live:
+        while True:
+            statuses = get_statuses()
+            finished_running = is_finished(statuses)
+
+            title = f"[green]COMPLETED[/]" if finished_running else f"[yellow]RUNNING...[/]"
+            table = Table(
+                title=title,
+                title_justify="left",
+                show_header=True,
+                header_style="bold",
+                box=None,
+            )
+            table.add_column("Resource Type")
+            table.add_column("Name (Variant)", width=50, no_wrap=True)
+            table.add_column("Status", width=10)
+            table.add_column("Error", style="red")
+
+            for status in statuses:
+                error = f" {status.error}" if status.error else ""
+                resource_type = status.resource_type.__name__
+                name = status.name
+                status_text = status.status
+
+                if status.resource_type is Provider:
```

Comment:
Could change this whole code block to:
status_text = status.status if status.resource_type is not Provider else "CREATED"
text_color = {
  "CREATED": "green",
  ...
}
status_type = text_color[status_text]

Reply:
good call. Yeah I didn't refactor what chatgpt spat out